### PR TITLE
Benchmark gym offloading

### PR DESCRIPTION
**Pull request by improbable-danny**
Please ignore commit history

### CODE REVIEW

File: Game/Source/GDKTestGyms/BenchmarkGymGameMode.h [hunk 1]
```diff
@@ -41,10 +41,13 @@ class GDKTESTGYMS_API ABenchmarkGymGameMode : public ABenchmarkGymGameModeBase
...
 	int32 PlayersSpawned;
 	TArray<AActor*> SpawnPoints;
-	TSubclassOf<APawn> NPCPawnClass;
 	TMap<int32, AActor*> PlayerIdToSpawnPointMap;
 	int32 NPCSToSpawn;
 
+	UPROPERTY()
+	ABenchmarkGymNPCSpawner* NPCSpawner;
+
+	bool ShouldUseCustomSpawning();
```
Comment (MatthewSandfordImprobable) [inline Game/Source/GDKTestGyms/BenchmarkGymGameMode.h:50]: ShouldUseCustomSpawning is this needed?

**Comment by MatthewSandfordImprobable** [comment]
Code looks good! Plan to get the blueprint reviewed?

File: Game/Source/GDKTestGyms/BenchmarkGymGameMode.h (hunk 1 above)
Comment (improbable-danny) [inline Game/Source/GDKTestGyms/BenchmarkGymGameMode.h:50]:
```suggestion

```

File: Game/Source/GDKTestGyms/BenchmarkGymGameMode.cpp [hunk 2]
```diff
@@ -248,7 +244,9 @@ void ABenchmarkGymGameMode::GenerateSpawnPoints(int CenterX, int CenterY, int Sp
 
 		const FVector SpawnLocation = FVector(X, Y, Z);
 		UE_LOG(LogBenchmarkGymGameMode, Log, TEXT("Creating a new PlayerStart at location %s."), *SpawnLocation.ToString());
-		SpawnPoints.Add(World->SpawnActor<APlayerStart>(APlayerStart::StaticClass(), SpawnLocation, FRotator::ZeroRotator, SpawnInfo));
+		AActor* Actor = World->SpawnActor<APlayerStart>(APlayerStart::StaticClass(), SpawnLocation, FRotator::ZeroRotator, SpawnInfo);
```
Comment (improbable-danny) [inline Game/Source/GDKTestGyms/BenchmarkGymGameMode.cpp:247]: fix this

File: Game/Source/GDKTestGyms/BenchmarkGymNPCSpawner.h [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+ */
+UCLASS()
+class GDKTESTGYMS_API ABenchmarkGymNPCSpawner : public AActor
+{
+	GENERATED_BODY()
+
+public:
+	UFUNCTION(Reliable, CrossServer)
+	void CrossServerSpawn(const FVector& SpawnLocation, const FBlackboardValues& BlackboardValues);
+	void CrossServerSpawn_Implementation(const FVector& SpawnLocation, const FBlackboardValues& BlackboardValues);
```
Comment (improbable-danny) [inline Game/Source/GDKTestGyms/BenchmarkGymNPCSpawner.h:24]:
```suggestion
```

File: Game/Source/GDKTestGyms/BenchmarkGymNPCSpawner.h [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+	ABenchmarkGymNPCSpawner();
+private:
+	TSubclassOf<APawn> NPCPawnClass;
```
Comment (improbable-danny) [inline Game/Source/GDKTestGyms/BenchmarkGymNPCSpawner.h:28]:
```suggestion
	UPROPERTY()
	TSubclassOf<APawn> NPCPawnClass;
```

File: Game/Source/GDKTestGyms/BenchmarkGymGameMode.cpp (hunk 2 above)
Comment (improbable-danny) [inline Game/Source/GDKTestGyms/BenchmarkGymGameMode.cpp:247]: SpawnPoints.Add(World->SpawnActor<APlayerStart>(APlayerStart::StaticClass(), SpawnLocation, FRotator::ZeroRotator, SpawnInfo));
```suggestion
		AActor* Actor = World->SpawnActor<APlayerStart>(APlayerStart::StaticClass(), SpawnLocation, FRotator::ZeroRotator, SpawnInfo);
```

File: Game/Source/GDKTestGyms/BenchmarkGymGameMode.cpp [hunk 5]
```diff
@@ -248,7 +244,9 @@ void ABenchmarkGymGameMode::GenerateSpawnPoints(int CenterX, int CenterY, int Sp
...
+		SpawnPoints.Add(Actor);
```
Comment (improbable-danny) [inline Game/Source/GDKTestGyms/BenchmarkGymGameMode.cpp:248]:
```suggestion
```

File: Game/Source/GDKTestGyms/BenchmarkGymGameMode.cpp [hunk 6]
```diff
@@ -248,7 +244,9 @@ void ABenchmarkGymGameMode::GenerateSpawnPoints(int CenterX, int CenterY, int Sp
...
+		Actor->SetReplicates(true);
```
Comment (improbable-danny) [inline Game/Source/GDKTestGyms/BenchmarkGymGameMode.cpp:249]:
```suggestion
```

**Pull request merged**